### PR TITLE
Integration tests &  N+1

### PR DESCRIPTION
**Pull request by KanzaTahreem**
### _Hey reviewer 👋_

In this PR, I did the following 🔽

- [x] Solved N+1 problem when fetching all posts and their comments for a user without using the `bullet gem`
- [x] Used `Capybara` to write integration tests for each view of the _Blog App_ project according to the [requirements](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md)
  
    - _Implemented tests for **_user index page_**_
    - _Implemented tests for **_user show page_**_
    - _Implemented tests for **_post index page_**_
    - _Implemented tests for **_post show page_**_

### _Before 👇_
n+1_before

### _After 👇_
n+1_after

### CODE REVIEW

File: spec/integration/post_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      @post = Post.create(title: 'First Post', text: 'This is my first post', author_id: @user.id)
+      @comment = Comment.create(text: 'First comment', author_id: @user.id, post_id: @post.id)
+      Like.create(author_id: @user.id, post_id: @post.id)
+
+      visit(user_posts_path(@user.id))
+    end
+
+    it "shows user's profile picture" do
+      expect(page).to have_css('img[src*="https://avatars.kanzatehreen.com"]')
+    end
```
Comment (Ugboaja-Uchechi) [inline spec/integration/post_index_spec.rb:20]: - [ ] Great job with the integration tests 👏. Unfortunately, `shows user's profile picture` test fails. Kindly fix this 🙏 

image

**Review by Ugboaja-Uchechi — CHANGES_REQUESTED** [review]
Hi @KanzaTahreem, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- Linters are passing ✔️ 
- README is descriptive 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/integration/post_index_spec.rb (hunk 1 above)
Comment (KanzaTahreem) [reply spec/integration/post_index_spec.rb:20]: Hey @Ugboaja-Uchechi 👋 Thank you for the review. 
But I have a doubt.  Can you please clear this one? All these tests are running in my pc.

**_Here's the screenshot for reference 🔽 That's for all tests_**

image

**_This is specifically for `post_index_page.rb` 🔽_**

image

**_Form my coding partner's system 🔽_**

image

File: spec/integration/post_index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,59 @@
+require 'rails_helper'
+
+RSpec.describe 'Post Index', type: :system do
```
Comment (GzimAsani) [inline spec/integration/post_index_spec.rb:3]: - All tests are passing, great job 
image

**Review by GzimAsani — APPROVED** [review]
# STATUS: APPROVED :100:         
## There is nothing else to say than it's time to merge... :partying_face: 
Cheers and Happy coding! :clap:  :computer:  :wine_glass: 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


## Optional suggestions


_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- NA
### Please be reminded to not open a new pull request after you get a review, only use the same one :thought_balloon: 


 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**